services/shuffle/neff/controller/action.go

```diff
@@ -29,7 +33,24 @@ func (a *initAction) Execute(ctx node.Context) error {
 		return xerrors.Errorf("failed to get signer: %v", err)
 	}
 
-	actor, err := neffShuffle.Listen(signer)
+	var service ordering.Service
+	err = ctx.Injector.Resolve(&service)
+	if err != nil {
+		return xerrors.Errorf("failed to resolve ordering.Service: %v", err)
+	}
+
+	var vs validation.Service
+	err = ctx.Injector.Resolve(&vs)
+	if err != nil {
+		return xerrors.Errorf("failed to resolve validation.Service: %v", err)
+	}
+
+	client := client{
+		srvc: service,
+		mgr:  vs,
+	}
```

Comment:
to keep this function short and sweet, I would consider splitting this in a dedicated function `makeClient(ctx node.Context) (client, err)`.